server/db/Cart.js

```diff
@@ -16,22 +17,32 @@ Cart.init(
   { sequelize: db, modelName: 'carts' }
 );
 
+// Finds the Cart and associated products, ordered by name
+Cart.getWithProducts = async (cartId) => {
+  try {
+    const cart = Cart.findByPk(cartId, {
+      include: {
+        model: CartProduct,
+        include: Product,
+        separate: true,
+        order: [[Product, 'name', 'ASC']],
+      },
+    });
+    return cart;
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+};
```

Comment:
In order to get the cart to order products by name, I had to change up how we build the carts on the backend, which ends up changing how the cart is stored on the frontend. Previously, we grabbed the cart like this:
```javascript
cart = {
  id: '...',
  products: [
    0: { id, name, ..., cartProducts: { quantity, ... } },
    1: { id, name, ..., cartProducts: { quantity, ... } },
  ]
}
```
Now, we grab them like this:
```javascript
cart = {
  id: '...',
  cartProducts: [
    0: { quantity: 2, product: { ... } },
    1: { quantity: 5, product: { ... } },
  ]
}
```

Reply:
I believe I've made all necessary changes to your Cart components, @vikkiwchan, but it's definitely worth a second set of eyes.